liblangutil/SourceReferenceFormatter.cpp

```diff
@@ -163,10 +173,15 @@ void SourceReferenceFormatter::printSourceLocation(SourceReference const& _ref)
 
 void SourceReferenceFormatter::printExceptionInformation(SourceReferenceExtractor::Message const& _msg)
 {
-	// exception header line
-	errorColored() << _msg.severity;
-	if (m_withErrorIds && _msg.errorId.has_value())
-		errorColored() << " (" << _msg.errorId.value().error << ")";
+    if (_msg.severity.eq("Error") && _msg.errorId.has_value())
+        errorColored() << " (" << _msg.errorId.value().error << ")";
+
+    else if(_msg.severity.eq("Warning") && _msg.errorId.has_value())
+        warningColored() << " (" << _msg.errorId.value().error << ")";
+
+    else if(_msg.severity.eq("Info") && _msg.errorId.has_value())
+        infoColored() << " (" << _msg.errorId.value().error << ")";
+
```

Comment:
If would be better to check `_msg.errorId.has_value()` first instead of repeating it in every condition.

Reply:
Fixed 👍